docs/src/main/asciidoc/security-architecture-concept.adoc

```diff
@@ -0,0 +1,38 @@
+////
+This document is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+[id="security-architecture-concept"]
+= Quarkus Security architecture
+include::_attributes.adoc[]
+:categories: security
+The Quarkus Security architecture provides several built-in authentication mechanisms. The `HttpAuthenticationMechanism` interface is the main entry mechanism for securing HTTP applications in Quarkus. Quarkus Security is also highly customizable.
+
+== Core components of Quarkus Security
+
+Quarkus Security uses `HttpAuthenticationMechanism` to extract the authentication credentials from the HTTP request and delegates them to `IdentityProvider` to convert the credentials to `SecurityIdentity`.
+For example, the credentials can come from the `Authorization` header, client HTTPS certificates, or cookies.
+
+`IdentityProvider` verifies the authentication credentials and maps them to `SecurityIdentity`, which has the username, roles, original authentication credentials, and other attributes.
+
+For every authenticated resource, you can inject a `SecurityIdentity` instance to get the authenticated identity information.
+
+In other contexts, it is possible to have other parallel representations of the same information or parts of it, for example, `SecurityContext` for JAX-RS or `JsonWebToken` for JSON Web Tokens (JWT).
+
+For more information, see xref:security-built-in-authentication-support.adoc[Built-in authentication support].
```

Comment:
We should also link to the authentication mechanism and identity provider concepts.
Not sure about  the built in authentication link being present in this section - since it is mostly about providing info about the Form/Basic/MTLS authentication

Reply:
@sberyozkin - Thanks :+1: 

 + 1 to suggestion to update the link in line 23.  Will also add links to those other 2 new topics. 

I think that most of the content in `security-built-in-authentication-support-concept.adoc` will soon recomposed so I held off changing the link until that piece of the work is done. 

The linked built-in topic will probably be retired and its current content moved over into several other more focussed diataxis topics, including:

- security-authentication-mechanisms.adoc
- security-overview-concept.adoc
- security-architecture-concept.adoc
and possibly a new concept topic for Proactive authentication, as there is a lot of content to warrant a separate concept (and possibly a how-to).

What do you think?

@rolfedh - cc - we overlap here on the diataxis restructuring of `security-built-in-authentication-support-concept.adoc`.

Reply:
@michelle-purcell 

>The linked built-in topic will probably be retired and its current content moved over into several other more focussed diataxis topics

Makes sense, `security-authentication-mechanisms` would be what `built-in auth`  is a concrete instance of, most of it . 

>and possibly a new concept topic for Proactive authentication, as there is a lot of content to warrant a separate concept

+1, I was just typing a similar suggestion and then noticed you had already thought about it :-)